lib/auth.js

```diff
@@ -1,5 +1,9 @@
 var asp = require('rsvp').denodeify;
 var request = require('request');
+if (!Promise) {
```

Comment:
Actually no, this will always have `Promise === undefined` since the `var Promise` below will hoist. So it is equivalent to just writing:

```javascript
var Promise = require('rsvp').Promise;
```